### ISSUE TITLE
no-unused-declarations: Unable to prune items from an object via destructuring

### ISSUE THREAD

**Issue by lukewlms** [cartant/tslint-etc#14]
At times we get a new, reduced object through destructuring and a rest parameter (see code below).  I'm getting an error in this case (`Unused declarations are forbidden`).

This shouldn't be an error - the TS language parser correctly determines that this is a valid case, including with --noUnusedLocals.  Can the rule be changed to allow this use case?

```
  const kitchenElements = {
    fork: true,
    spoon: true,
    cup: true
  }

  const {fork, ...notSharpThings} = kitchenElements;

  console.log(notSharpThings);
```

Thank you!

**Comment by cartant** [comment]
Yeah. Any false positives should be fixable.

This is one rule in which creating fixtures that cover all possible use cases is pretty much impractical. There are plenty [in the fixtures](https://github.com/cartant/tslint-etc/tree/master/fixtures/no-unused-declaration), but they won't cover everything.

I can look at fixing this later. Given that you've included a snippet that effects the false positive, fixing this should be straightforward. Thanks for the bug report.

**Comment by cartant** [comment]
I re-read your issue. I see what you are talking about. This is something that's bitten me before, too. Yeah. I agree that if a rest property is used in object descructuring, the other properties should count as used.

**Comment by cartant** [comment]
This should be fixed in 1.4.3.
